packages/recorder/src/index.ts

```diff
@@ -3,21 +3,53 @@ import {SessionManager} from './session_recording/sessions_manager';
 import {initTerminal} from './terminal_interaction';
 import {AppDependencies} from './types';
 
-const defaultDestination = process.env.DESTINATION;
-const useDestinationHostHeader = process.env.USE_DESTINATION_HOST_HEADER;
+import {Command} from 'commander';
+const program = new Command();
 
-if (!defaultDestination && useDestinationHostHeader !== 'true') {
-    console.log('Please provide a destination via the DESTINATION enviornment variable, or bypass this check by setting the USE_DESTINATION_HOST_HEADER bool environment variable');
+const runServerForDestination = async (host: string, port: string, deps: AppDependencies) => {
+    const app = initApp({
+        ...deps,
+        defaultDestination: host,
+    });
+
+    return new Promise<void>(r => {
+        app.listen(port, () => {
+            console.log(`http://localhost:${port}`);
+            setTimeout(() => {
+                r();
+            }, 50);
+        });
+    });
 }
 
-const sessionManager = new SessionManager();
-const deps: AppDependencies = {sessionManager, defaultDestination};
+program
+    .name('fc-record')
+    .description('CLI to record HTTP request sessions')
+    .version('0.0.1');
+
+// Usage: Destination host and local port pairs. A vertical bar is used to join the destination host and local port. The destinations are separated by spaces.
+program.command('record')
+    .description('Record requests')
+    .option('-d, --destinations [host|port...]', '')
+    .action(async (parsed: {destinations: string[]}, options) => {
+        const destinations = parsed.destinations;
```

Comment:
In what way could these be hardened? What is it about them that is potentially error prone?

Reply:
The code below this comment (basically this whole function here) is where the issues are. We're making certain assumptions about the data provided by the user, specifically we're assuming the provided destinations are in the format of `https://google.com|3000`, which is a host to proxy to, followed by a vertical bar `|`, followed by a local port to use to proxy the requests. The code should first verify that the provided values are formed correctly before proceeding with using the values.

In the case that there is malformed arguments provided, we should log a message and exit the program, rather than waiting for an actual error to happen because of the malformed arguments.

Reply:
I think when I said "the code below this comment", you may have thought I meant "the line of code below this comment", when I actually vaguely meant "somewhere within the next 10-20 lines in the block of code that is below this comment"